src/Akka.Streams.Kafka/Stages/CommittableConsumerStage.cs

```diff
@@ -101,58 +107,39 @@ public override void PreStart()
                     break;
             }
 
-            _messagesReceived = GetAsyncCallback<ConsumeResult<K, V>>(MessagesReceived);
             _partitionsAssigned = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsAssigned);
             _partitionsRevoked = GetAsyncCallback<IEnumerable<TopicPartitionOffset>>(PartitionsRevoked);
-            ScheduleRepeatedly(TimerKey, _settings.PollInterval);
         }
 
         public override void PostStop()
         {
             Log.Debug($"Consumer stopped: {_consumer.Name}");
             _consumer.Dispose();
+            _completion.SetResult(NotUsed.Instance);
 
             base.PostStop();
         }
         
-        protected override void OnTimer(object timerKey) => PullQueue();
-
         //
         // Consumer's events
         //
 
-        private void HandleMessage(ConsumeResult<K, V> message) => _messagesReceived(message);
-        
-        private void MessagesReceived(ConsumeResult<K, V> message)
+        private void HandleOnPartitionsAssigned(IConsumer<K, V> consumer, List<TopicPartition> list)
```

Comment:
When does this method get assigned?

Reply:
Sorry, did not get it: which method do you mean? `HandleMessage` method is not needed anymore, because we are not buffering right now and just pushing message to the downstream.
And `HandleOnPartitionsAssigned` method is used as a callback for `partitionsAssigned` consumer event, it is calling "_partitionsAssigned" thread-safe delegate and that delegate is a wrapper for `PartitionsAssigned` method. Like this:

```csharp
public override void PreStart()
{
	// ...
	_consumer = _settings.CreateKafkaConsumer(..., HandlePartitionsAssigned, ...);
	// ...
	_partitionsAssigned = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsAssigned);
	// ...
}


private void HandlePartitionsAssigned(IConsumer<K, V> consumer, List<TopicPartition> list)
{
	_partitionsAssigned(list);
}

private void PartitionsAssigned(IEnumerable<TopicPartition> partitions)
{
	// Real handling is implemented here
}
```

Reply:
Ahhh I see.